.github/workflows/links-fail-fast.yml

```diff
@@ -7,6 +7,8 @@ on:
 jobs:
   check-links:
     runs-on: ubuntu-latest
+    permissions:
+      contents: read
```

Comment:
Given this only needs read permissions, can we use an alternate token than the `GITHUB_TOKEN` which has more privileges than that?

Reply:
Provided the org is set as restrictive access, the only other data the token can retrieve is metadata and packages with read-only. Nothing can be written.
https://docs.github.com/en/actions/security-for-github-actions/security-guides/automatic-token-authentication#permissions-for-the-github_token

So I don't really see what other privileges you're talking about.
The advantage of using the auto GITHUB_TOKEN is also that it's handled automatically, and doesn't require a bot account.

Reply:
> So I don't really see what other privileges you're talking about.

![20241002_092832](https://github.com/user-attachments/assets/c7f1ee00-0c47-424e-9472-aa80e5b66d48)

Reply:
> You can use the permissions key in your workflow file to modify permissions for the GITHUB_TOKEN for an entire workflow or for individual jobs. This allows you to configure the minimum required permissions for a workflow or job. When the permissions key is used, all unspecified permissions are set to no access, with the exception of the metadata scope, which always gets read access.

Based on this, even if the org is indeed configured as "permissive", since we're setting the `permissions` key, access we're not setting will be set to no access. So only `contents` and `metadata` will be available.